Zend/zend_hash.c

```diff
@@ -2523,7 +2523,7 @@ ZEND_API HashTable* ZEND_FASTCALL zend_symtable_to_proptable(HashTable *ht)
 					Z_ADDREF_P(zv);
 				}
 			} while (0);
-			zend_hash_add_new(new_ht, str_key, zv);
+			zend_hash_add(new_ht, str_key, zv);
```

Comment:
`zend_hash_update` overwrites, while `zend_hash_add` does not. Either is fine generally, but if you use `zend_hash_add` you also need to perform error handling for the case where the element already exists.